lib/client.js

```diff
@@ -247,8 +248,8 @@ class Client extends Dispatcher {
       this[kQueue].push(request)
       if (this[kResuming]) {
         // Do nothing.
-      } else if (util.isStream(request.body)) {
-        // Wait a tick in case stream is ended in the same tick.
+      } else if (util.isStream(request.body) || util.isStrictIterable(request.body)) {
```

Comment:
This is not needed for iterators

Reply:
I thought so as well, but there's a test the fails without this. However, it an async-iterator that wraps a stream so it might be related. I'll take another look.

Reply:
Specifically, `pipelining idempotent busy async-iterator` (`pipeliningIdempotentBusy` in `client-pipelining.js`) doesn't pass.

Reply:
I guess we need it then.